cEP-0015.md

```diff
@@ -0,0 +1,98 @@
+# corobo Enhancement (security, tests and configurability)
+
+| Metadata |                                                          |
+| -------- | -------------------------------------------------------- |
+| cEP      | 15                                                       |
+| Version  | 1.0                                                      |
+| Title    | corobo Enhancement (security, tests and configurability) |
+| Authors  | Nitanshu Vashistha <mailto:nitanshu.vzard@gmail.com>     |
+| Status   | Proposed                                                 |
+| Type     | Process                                                  |
+
+## Abstract
+
+This cEP describes the details of enhancement of
+[corobo](https://github.com/coala/corobo) in terms of security, tests,
+configurability and the new plugins that are to be added to corobo as part
+of the
+[GSoC Project](https://summerofcode.withgoogle.com/projects/#6603667076022272).
+
+## Security
+
+### Why improve security?
+
+Security has been one of the major concerns due to some past experiences.
+We want to provide access controls based on user’s GitHub team memberships.
+
+### How to improve security?
+
+1. Make all LabHub commands except invite require being a member of the
+   organization.
+
+2. Add ability to ban from all gitter rooms at once.
+
+3. Remove auto-invite and give invite permissions to developers.
+
+4. Prevent newcomers from self-assigning themselves and give developers
+   permission to assign newcomers.
+
+5. Prevent corobo being used to spam a room.
+
+6. Force newcomers to finish one issue first.
+
+7. Require newcomers to find a newcomer issue to work on before they are
```

Comment:
so like looking through newcomer issues and checking for possibilities like:
* no ones assigned to it
* no one claimed it in the last week
* no one pushed changes or was even active on gitter since the last week, in which case atleast newcomer issues can be reassigned as they don't require more work than that, also pinging newcomers of their unassignment that they havent worked for so and so time on a newcomer issue and it **will** be reassigned, more time gap for low issues
* discarding issues which have users waiting in the queue wanting to get assigned both from gitter and comments on GitHub 

this stuff would be cool af,
would love to see it actually working

Reply:
@ishanSrt  Thanks for the suggestion but I guess this would be GitMate type of stuff and might not be possible to write a plugin for this.
@Vamshi99 what do you say about this suggestion?